core/src/modules/apiData.ts

```diff
@@ -1,4 +1,51 @@
 export namespace APIData {
+  export function ensureObject(
+    param: { [key: string]: any } | string,
+    recursing = false
+  ) {
+    if (!param) {
+      return null;
+    } else if (Array.isArray(param)) {
+      try {
+        return param.map((row) => APIData.ensureObject(row, true));
+      } catch (error) {
+        throw new Error(
+          `${param} cannot be converted to JSON object (${error})`
+        );
+      }
+    } else if (typeof param === "string") {
+      try {
+        return JSON.parse(param) as { [key: string]: any };
+      } catch (error) {
+        if (recursing) {
+          return param;
+        } else {
+          throw new Error(
+            `${param} cannot be converted to JSON object (${error})`
+          );
+        }
+      }
+    } else {
+      return param;
+    }
+  }
+
+  export function ensureBoolean(param: boolean | string | number) {
+    if (param === true || param === false) return param;
+    if (param === 1 || param === "1" || param === "true") return true;
+    if (param === 0 || param === "0" || param === "false") return false;
+    throw new Error(`${param} cannot be converted to a boolean`);
+  }
+
+  export function ensureNumber(param: string | number) {
+    if (typeof param === "number") return param;
+    try {
+      return parseFloat(param);
```

Comment:
Good question! 

I made the parser look for floats rather than integer because I wanted to reflect what the user had intended at this stage.  If limit/offset fails because the user sent a decimal, that would throw an exception elsewhere - probably by Seqeulzie

Also... it turns out decimal limits/offsets are valid in postgres maybe?
<img width="690" alt="Screen Shot 2021-08-19 at 5 27 27 PM" src="https://user-images.githubusercontent.com/303226/130160608-677a7224-5325-4469-80d4-e351917de715.png">